components/WeeklyOverview/index.js

```diff
@@ -58,28 +63,31 @@ export default function WeeklyOverview() {
       ...prevStatus,
       [weekday]: {
         ...prevStatus[weekday],
-        [supplementKey]: false,
+        [supplementKey]: {},
       },
     }));
   };
 
-  const handleCheckSupplement = (weekday, supplementKey) => {
+  const handleCheckSupplement = (weekday, supplementKey, selectedTime) => {
     setSupplementStatus((prevStatus) => ({
       ...prevStatus,
       [weekday]: {
         ...prevStatus[weekday],
-        [supplementKey]: !prevStatus[weekday][supplementKey],
+        [supplementKey]: {
+          ...prevStatus[weekday][supplementKey],
+          [selectedTime]: !prevStatus[weekday]?.[supplementKey]?.[selectedTime] || false,
+        },
       },
     }));
   };
 
-  const handleDeleteMedication = (weekday, medicationIndex) => {
-    const updatedMedications = supplementsByWeekday[weekday].filter(
-      (_, index) => index !== medicationIndex
-    );
-    setSupplementsByWeekday((prevSupplements) => ({
-      ...prevSupplements,
-      [weekday]: updatedMedications,
+  const handleTimeSelection = (weekday, supplementKey, selectedTime) => {
```

Comment:
you don't work with the supplementKey here, so you can remove this parameter:

```suggestion
  const handleTimeSelection = (weekday, selectedTime) => {
```